lib/app/utils/search_validation.rb

```diff
@@ -0,0 +1,139 @@
+# frozen_string_literal: true
+
+module Inferno
+  module SearchValidationUtil
+    def can_resolve_path(element, path)
```

Comment:
I think we should leave this and `resolve_element_from_path` in `SequenceBase` for now since they're general methods which are used in contexts other than search date validation. I suspect that these might make the most sense as extensions on FHIR models so that we can do something like `resource.resolve? some_path`, but that is outside the scope of this PR.